src/nyc_trees/js/src/adminTerritoryPage.js

```diff
@@ -86,5 +133,49 @@ function onBlockfaceAdd(gridData) {
 }
 
 function onBlockfaceRemove(feature) {
+    delete selectedBlockfaces[feature.properties.id];
     return true;
-}
\ No newline at end of file
+}
+
+var drawer = null,
+    willAddBlocks = null;
+
+function drawAreaToAdd() { drawArea(true); }
+function drawAreaToRemove() { drawArea(false); }
+
+function drawArea(willAdd) {
+    willAddBlocks = willAdd;
+    toggleAreaControls();
+    selectedBlockfaces.clicksEnabled = false;
```

Comment:
This is in fact a problem, and using `selectedLayer` fixed it. Thanks for the eagle-eye type checking.